middleware/hosts/hosts.go

```diff
@@ -49,17 +50,21 @@ func (h Hosts) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (
 		answers = h.ptr(qname, names)
 	case dns.TypeA:
 		ips := h.LookupStaticHostV4(qname)
+		otherRecordCount = len(h.LookupStaticHostV6(qname))
 		answers = a(qname, ips)
 	case dns.TypeAAAA:
 		ips := h.LookupStaticHostV6(qname)
+		otherRecordCount = len(h.LookupStaticHostV4(qname))
 		answers = aaaa(qname, ips)
 	}
 
 	if len(answers) == 0 {
 		if h.Fallthrough {
 			return middleware.NextOrFailure(h.Name(), h.Next, ctx, w, r)
 		}
-		return dns.RcodeRefused, nil
+		if otherRecordCount == 0 {
+			return dns.RcodeRefused, nil
```

Comment:
The hosts middleware is more like the auto middleware, in that the origin is generally defined as "."
We wanted to return a REFUSED instead of an NXDOMAIN since we don't want to say we are authoritative for that zone and we don't necessarily want it cached.
I am not necessarily married to that decision though, and if it will be better to return an NXDOMAIN I can go ahead and change that.